Fixed HTTP status return code and added client IP address

dleborgne opened the pull request.

dleborgne committed:
Fixed HTTP status code

dleborgne committed:
Populate ip field in database

dleborgne committed:
New getIP() function returns client IP address

dleborgne committed:
Add IP address to database records

dleborgne committed:
Add ip field in csp_report table

dleborgne committed:
Fixed missing bind_param variable

o- commented:
Thanks for the patch, but it was a conscious decision not to log the ip address here! The csp notification mechanism is already problematic enough from a privacy perspective. I don't think its a good example to log more data than necessary here, or users will just disable it en-masse (rightfully so). 
What was your usecase (ie. your reason to log that IP), that made you write this patch? Maybe there is a less invasive way to achieve it?

dleborgne committed:
Fixed missing parenthesis

dleborgne committed:
Better hande proxied requests 

If available, use HTTP_X_FORWARDED_FOR instead of REMOTE_ADDR to get client ip address
